contrib/clojure-package/src/org/apache/clojure_mxnet/ndarray.clj

```diff
@@ -167,3 +171,36 @@
 
 (defn shape-vec [ndarray]
   (mx-shape/->vec (shape ndarray)))
+
+(s/fdef vec->nd-vec
+        :args (s/cat :v vector? :shape-vec vector?)
+        :ret vector?)
+
+(defn- vec->nd-vec
+  "Convert a vector `v` into a n-dimensional vector given the `shape-vec`
+
+   Ex:
+     (vec->nd-vec [1 2 3] [1 1 3])       ;[[[1 2 3]]]
+     (vec->nd-vec [1 2 3 4 5 6] [2 3 1]) ;[[[1] [2] [3]] [[4] [5] [6]]]
+     (vec->nd-vec [1 2 3 4 5 6] [1 2 3]) ;[[[1 2 3]] [4 5 6]]]
+     (vec->nd-vec [1 2 3 4 5 6] [3 1 2]) ;[[[1 2]] [[3 4]] [[5 6]]]
+     (vec->nd-vec [1 2 3 4 5 6] [3 2])   ;[[1 2] [3 4] [5 6]]"
+  [v [s1 & ss :as shape-vec]]
+  (if-not (seq ss)
+    (vec v)
+    (->> v
+         (partition (/ (count v) s1))
```

Comment:
You will need to use `clojure.core//` instead of just `/` because the ndarray has a definition for the operator in this namespace

Reply:
good catch!